### ISSUE TITLE
CMSPAR bit is never clear when back to other parity setting.

### ISSUE THREAD

**Issue by hetii** [pyserial/pyserial#179]
In serial/serialposix.py there is bug regarding CMSPAR bit.
So its possible to set MARK or SPACE parity, but after that its not possible to back to other parity modes. This bit remains always set.

        # setup parity
        iflag &= ~(termios.INPCK | termios.ISTRIP)
        if self._parity == serial.PARITY_NONE:
            cflag &= ~(termios.PARENB | termios.PARODD)
        elif self._parity == serial.PARITY_EVEN:
            cflag &= ~(termios.PARODD)
            cflag |= (termios.PARENB)
        elif self._parity == serial.PARITY_ODD:
            cflag |= (termios.PARENB | termios.PARODD)
        elif self._parity == serial.PARITY_MARK and plat[:5] == 'linux':
            cflag |= (termios.PARENB | CMSPAR | termios.PARODD)
        elif self._parity == serial.PARITY_SPACE and plat[:5] == 'linux':
            cflag |= (termios.PARENB | CMSPAR)
            cflag &= ~(termios.PARODD)
        else:
            raise ValueError('Invalid parity: %r' % self._parity)

**Comment by zsquareplusc** [comment]
That's right, but the quoted code is not from the current version, which has this fixed:
https://github.com/pyserial/pyserial/blob/master/serial/serialposix.py#L371
(`CMSPAR` defaults to 0 and is only set for Linux)

**Comment by damoyelang1992** [comment]
I think add these code will fix it, but i do not know if it is right:
`
from serial import serialposix
serialposix.CMSPAR = 1
`
